os/board/rtl8730e/src/rtl8730e_st7701.c

```diff
@@ -0,0 +1,202 @@
+		LCDC_InitStruct.layerx[idx].LCDC_LayerColorKeyingEn = ENABLE;
+	}
+}
+
+static u32 LCDCRgbHvIsr(void *Data)
+{
+	LCDC_TypeDef *LCDCx = (LCDC_TypeDef *)Data;
+	volatile u32 IntId = LCDC_GetINTStatus(LCDCx);
+
+	if (IntId & LCDC_BIT_LCD_LIN_INTS) {
+		LCDC_ClearINT(LCDCx, LCDC_BIT_LCD_LIN_INTS);
+	}
+	if (IntId & LCDC_BIT_DMA_UN_INTS) {
+		LCDC_ClearINT(LCDCx, LCDC_BIT_DMA_UN_INTS);
+		UnderFlowCnt++;
+		if (UnderFlowCnt == 1) {
+			lldbg("LCDC DMA Under-----\n");
+		}
+	}
+	return 0;
+}
+
+static int rtl8730_st7701_lcd_init(void) {
+
+	//MIPI_DSI_INT_Config(MIPI, DISABLE, DISABLE, FALSE);
+	LcdcInitValues();
+
+	rtl8730_st7701_lcd_layer_enable(0, true);
+	rtl8730_st7701_lcd_layer_enable(1, false);
+	rtl8730_st7701_lcd_layer_enable(2, false);
+
+	LCDC_Init(LCDC, &LCDC_InitStruct);
+	LCDC_DMAModeConfig(LCDC, LCDC_LAYER_BURSTSIZE_4X64BYTES);
+	LCDC_DMADebugConfig(LCDC, LCDC_DMA_OUT_DISABLE, 0);
+
+	LCDC_LineINTPosConfig(LCDC, YRES * 4 / 5);
+	LCDC_INTConfig(LCDC, LCDC_BIT_LCD_LIN_INTEN | LCDC_BIT_DMA_UN_INTEN, ENABLE);	
+}
+
+
+static int rtl8730_st7701_gpio_reset(void) {
+}
+
+static int rtl8730_st7701_cache_invalidate(u32 *buffer, int size) {
+	DCache_CleanInvalidate((u32)buffer, size); 
+}
+
+static int rtl8730_st7701_lcd_reload(void) {
+	LCDC_TrigerSHWReload(pLCDC);
+}
+
+
+static int rtl8730_st7701_lcd_enable(void) {
+	LcdcEnable();
+}
+
+static int rtl8730_st7701_lcd_layer_enable(int layer, bool enable) {
+	if(enable){
+		LCDC_InitStruct.layerx[layer].LCDC_LayerEn = ENABLE;
+	}
+	else{
+		LCDC_InitStruct.layerx[layer].LCDC_LayerEn = DISABLE;
+	}
+}
+
+static void lcd_put_area(u8 *LcdImgBuffer, u32 x1, u32 y1, u32 x2, u32 y2){
+
+	//lldbg("data :%x %x\n", LcdImgBuffer[0], LcdImgBuffer[1]);
+	LCDC_InitStruct.layerx[0].LCDC_LayerImgBaseAddr = (u32)LcdImgBuffer;
+	/*LCDC_InitStruct.layerx[0].LCDC_LayerHorizontalStart = x1;
+	LCDC_InitStruct.layerx[0].LCDC_LayerHorizontalStop = x2;
+	LCDC_InitStruct.layerx[0].LCDC_LayerVerticalStart = y1;
+	LCDC_InitStruct.layerx[0].LCDC_LayerVerticalStop = y2;*/
+
+	LCDC_LayerConfig(LCDC, 0, &LCDC_InitStruct.layerx[0]);
+	LCDC_TrigerSHWReload(LCDC);
+}
+
+static void LcdcEnable(void)
+{
+	LCDC_TypeDef *pLCDC = LCDC;
+
+	/*enable the LCDC*/
+	LCDC_Cmd(pLCDC, ENABLE);
+	while (!LCDC_CheckLCDCReady(pLCDC));
+
+	MIPI_DSI_Mode_Switch(MIPI, ENABLE);
+}
+
+void rtl8730_st7701_initialize(void)
+{
+	struct lcd_data config;
+	config.XPixels = XRES;
+	config.YPixels = YRES;
+	
+	FAR struct mipi_dsi_host *dsi_host = mipi_dsi_host_initialize(&config);
+	FAR struct mipi_dsi_device *dsi_device = st7710_dsi_device_initialize(dsi_host);
+	//FAR struct mipi_dsi_device *dsi_device = mipi_dsi_device_register(dsi_host, "dsi", 0);
+	struct lcd_dev_s *dev = st7701_lcdinitialize(dsi_device, &g_rtl8730_st7701_config_s);
+	rtl8730_st7701_lcd_init();
+
+	if (lcddev_register(dev) < 0) {
+		lldbg("ERROR: LCD driver register fail\n");
+	} else {
+		lldbg("LCD driver register success\n");
+	}
+	LcdcEnable();
+	return;
+}
```

Comment:
Could you check the warning `no line end`?

Reply:
can you please clarify, what's "no line end" warning.